python/ray/dataframe/dataframe.py

```diff
@@ -1502,6 +1527,24 @@ def iloc(axis=None):
         raise NotImplementedError("Not Yet implemented.")
 
 
+def _get_lengths(df):
+    """Gets the length of the dataframe.
+
+    Args:
+        df: A remote pd.DataFrame object.
+
+    Returns:
+        Returns an integer length of the dataframe object. If the attempt
+            fails, returns 0 as the length.
+    """
+    try:
+        return len(df)
+    # Because we sometimes have cases where we have summary statistics in our
+    # DataFrames
+    except TypeError:
+        return 0
```

Comment:
Should we be returning 0 or letting the exception happen?

Reply:
Since it happens automatically, we need to suppress the error. There are cases where we don't have a `__len__` or `size`  in our `DataFrame` because we are wrapping anything that gets returned (e.g. `sum`). In the future, we will change this so there's no ambiguity with what a `DataFrame` can return.

Reply:
got it